index.d.ts

```diff
@@ -13,15 +13,15 @@ interface LocalizationContextValue {
 }
 
 interface LocalizationProviderProps {
-  locale: string;
+  locale?: string;
   disableCache?: boolean;
   translations: Record<string, unknown>;
 }
 
 interface MessageComponentProps {
   descriptor: string;
-  values: Record<string, unknown>;
-  defaultMessage: string;
+  values?: Record<string, unknown>;
```

Comment:
Let's declare and use interfaces for values and translations. Basically the types are not `unknown`:
```js
values?: Record<string, string | number>;

// ....

interface Translations {
  [key: string]: string | number | Translations;
}
```